corehq/messaging/scheduling/static/scheduling/js/create_schedule.ko.js

```diff
@@ -3,14 +3,153 @@ hqDefine("scheduling/js/create_schedule.ko", function() {
+            } else if(newValue === 'weekly') {
+                occurrences.text(gettext("After weeks:"));
+            } else if(newValue === 'monthly') {
+                occurrences.text(gettext("After months:"));
+            }
+        };
+
+        self.send_frequency.subscribe(self.setOccurrencesOptionText);
+
+        self.showTimeInput = ko.computed(function() {
+            return self.send_frequency() !== 'immediately';
+        });
+
+        self.showStartDateInput = ko.computed(function() {
+            return self.send_frequency() !== 'immediately';
+        });
+
+        self.showWeekdaysInput = ko.computed(function() {
+            return self.send_frequency() === 'weekly';
+        });
+
+        self.showDaysOfMonthInput = ko.computed(function() {
+            return self.send_frequency() === 'monthly';
+        });
+
+        self.showStopInput = ko.computed(function() {
+            return self.send_frequency() !== 'immediately';
+        });
+
+        self.computedEndDate = ko.computed(function() {
+            if(self.stop_type() !== 'never') {
+                var start_date_milliseconds = Date.parse(self.start_date());
+                var occurrences = parseInt(self.occurrences());
+
+                if(!isNaN(start_date_milliseconds) && !isNaN(occurrences)) {
+                    var milliseconds_in_a_day = 24 * 60 * 60 * 1000;
+                    if(self.send_frequency() === 'daily') {
+                        var end_date = new Date(start_date_milliseconds + ((occurrences - 1) * milliseconds_in_a_day));
+                        return end_date.toJSON().substr(0, 10);
+                    } else if(self.send_frequency() === 'weekly') {
+                        var js_start_day_of_week = new Date(start_date_milliseconds).getUTCDay();
+                        var python_start_day_of_week = (js_start_day_of_week + 6) % 7;
+                        var offset_to_last_weekday_in_schedule = null;
+                        for(var i = 0; i < 7; i++) {
+                            var current_weekday = (python_start_day_of_week + i) % 7;
+                            if(self.weekdays().indexOf(current_weekday.toString()) !== -1) {
+                                offset_to_last_weekday_in_schedule = i;
+                            }
+                        }
+                        if(offset_to_last_weekday_in_schedule !== null) {
+                            var end_date = new Date(
+                                start_date_milliseconds +
+                                (occurrences - 1) * 7 * milliseconds_in_a_day +
+                                offset_to_last_weekday_in_schedule * milliseconds_in_a_day
+                            );
+                            return end_date.toJSON().substr(0, 10);
+                        }
+                    } else if(self.send_frequency() === 'monthly') {
+                        var last_day = null;
+                        self.days_of_month().map(function(value) {
```

Comment:
nitpick, but `forEach` seems more logical here since you're not doing anything with the return value of `map`

Reply:
makes sense, https://github.com/dimagi/commcare-hq/pull/18495/commits/dcfbf6beb9cf569109ed1991942fbf2e6c29b1c3